Source/Scene/GlobeSurfaceTileProvider.js

```diff
@@ -438,10 +440,16 @@ define([
      * @returns {Visibility} The visibility of the tile.
      */
     GlobeSurfaceTileProvider.prototype.computeTileVisibility = function(tile, frameState, occluders) {
-        var surfaceTile = tile.data;
+        if (frameState.fogEnabled) {
+            var distance = this.computeDistanceToTile(tile, frameState);
```

Comment:
We are now computing this distance more than once per frame, right?  It burns a good bit of CPU (profile, but I bet it is non-trivial).  Let's cache this like 3D Tiles: https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles/Source/Scene/Cesium3DTileset.js#L389